core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -54,6 +60,7 @@ public class SharePartition {
      * The state of the records determines if the records should be re-delivered, move the next fetch
      * offset, or be state persisted to disk.
      */
+    // Visible for testing
     public enum RecordState {
```

Comment:
Can we make it `default` instead of `public`

Reply:
thanks, done. I agree.